app/src/main/java/app/revanced/integrations/utils/LogHelper.java

```diff
@@ -64,58 +74,50 @@ public static void printDebug(LogMessage message) {
+     * Logs exceptions under the outer class name of the code calling this method.
      */
-    public static void printException(LogMessage message, Throwable ex) {
-        Log.e("revanced: " + message.findOuterClassSimpleName(), message.buildMessageString(), ex);
+    public static void printException(@NonNull LogMessage message) {
+        printException(message, null, null);
     }
 
     /**
-     * Deprecated. Instead call {@link #printDebug(LogMessage)},
-     * which does not cause log messages to be constructed unless logging is enabled.
+     * Logs exceptions under the outer class name of the code calling this method.
      */
-    @Deprecated
-    public static void debug(Class _clazz, String message) {
-        printDebug(() -> message); // this fails to show the correct calling class name, but it's deprecated who cares
+    public static void printException(@NonNull LogMessage message, @Nullable Throwable ex) {
+        printException(message, ex, null);
     }
 
     /**
-     * Deprecated.  Instead call {@link #printException(LogMessage, Throwable)}
-     * or {@link #printException(LogMessage)}
-     * which does not cause log messages to be constructed unless logging is enabled.
+     * Logs exceptions under the outer class name of the code calling this method.
+     *
+     * @param message          log message
+     * @param ex               exception (optional)
+     * @param userToastMessage user specific toast message to show instead of the log message (optional)
      */
-    @Deprecated
-    public static void printException(Class _clazz, String message, Throwable ex) {
-        printException(() -> message, ex);
-    }
-
-    /**
-     * Deprecated. Instead call {@link #printException(LogMessage)},
-     * which does not cause log messages to be constructed unless logging is enabled.
-     */
-    @Deprecated
-    public static void printException(Class _clazz, String message) {
-        printException(() -> message);
+    public static void printException(@NonNull LogMessage message, @Nullable Throwable ex,
+                                      @Nullable String userToastMessage) {
+        String messageString = message.buildMessageString();
+        String outerClassSimpleName = message.findOuterClassSimpleName();
+        String logMessage = REVANCED_LOG_PREFIX + outerClassSimpleName;
+        if (ex == null) {
+            Log.e(logMessage, messageString);
+        } else {
+            Log.e(logMessage, messageString, ex);
+        }
+        ReVancedUtils.runOnMainThread(() -> {
+            Context context = ReVancedUtils.getContext();
+            if (context != null) {
+                String toastMessageToDisplay = (userToastMessage != null)
+                        ? userToastMessage
+                        : outerClassSimpleName + ": " + messageString;
+                Toast.makeText(context, toastMessageToDisplay, Toast.LENGTH_LONG).show();
+            }
+        });
```

Comment:
This might be unwanted or at least should be toggleable. Imagine the RYD servers are down and you receive an annoying toast on each video.

Reply:
Connection errors are the only realistic reason anyone would ever turn the toasts off.

I restored the setting switch, if someone really wants the toasts gone.

Reply:
Any other exception which causes the toast to show up multiple times in short periods is reason to as well.